ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -527,13 +527,59 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+public func switchToLatest<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+	return SignalProducer<T, E> { sink, disposable in
+		let producerCompleted = Atomic(false)
+		let latestCompleted = Atomic(false)
```

Comment:
I'm not sure if this would make the code more reasonable, but this could be done with a single boolean, say `canComplete`, I think.